File: web/src/services/user-service.ts
import { api } from "./api";

import type { AppUser, UserCreatePayload, UserUpdatePayload } from "@/types/user";

interface UserFilters {
  role?: string;
  tenantId?: number | null;
  branchId?: number | null;
}

export async function getUsers(filters: UserFilters = {}) {
  const response = await api.get("/users", {
    params: {
      role: filters.role,
      tenant_id: filters.tenantId ?? undefined,
      branch_id: filters.branchId ?? undefined,
    },
  });

  return response.data.data as AppUser[];
}

export async function createUser(payload: UserCreatePayload) {
  const response = await api.post("/users", payload);
  return response.data.data as AppUser;
}

export async function updateUser(userId: number, payload: UserUpdatePayload) {
  const response = await api.put(`/users/${userId}`, payload);
  return response.data.data as AppUser;
}

export async function deleteUser(userId: number) {
  const response = await api.delete(`/users/${userId}`);
  return response.data;
}

File: api/app/services/user_service.py
from app.repositories.user_repository import (
    UserRepository
)


class UserService:

    @staticmethod
    def get_users(
        role: str = None,
        tenant_id: int = None,
        branch_id: int = None
    ):
        return UserRepository.find_filtered(
            role,
            tenant_id,
            branch_id
        )

    @staticmethod
    def get_user_by_id(user_id: int):
        return UserRepository.find_by_id(user_id)

    @staticmethod
    def get_raw_user_by_email(email: str):
        return UserRepository.find_raw_by_email(email)

    @staticmethod
    def create_user(payload):
        return UserRepository.create(payload)

    @staticmethod
    def update_user(
        user_id: int,
        payload
    ):
        return UserRepository.update(
            user_id,
            payload
        )

    @staticmethod
    def delete_user(user_id: int):
        return UserRepository.delete(user_id)
